one_fm/one_fm/utils.py

```diff
@@ -266,4 +267,132 @@ def notify_operator_with_supervisor_response(name):
+    '''
+        Method used to create compensatory leave request and additional salary for holiday attendance
+        on_submit or on_cancel of attendance will trigger this method
+        args:
+            doc is attendance object
+            method is the method from the hook (like on_submit)
+        if method is "on_submit", then create and submit the compensatory leave request and additional salary
+        if method is "on_cancel", then cancel the compensatory leave request and additional salary, that is created for the attendance
+    '''
+
+    # Get holiday list of dicts with `holiday_date` and `description`
+    holidays = get_holidays_for_employee(doc.employee, doc.attendance_date, doc.attendance_date)
+
+    # process compensatory leave request and additional salary if attendance status not equals "Absent" or "On Leave"
+    if len(holidays) > 0 and doc.status not in ["Absent", "On Leave"]:
+        salary_component = frappe.db.get_single_value('HR and Payroll Additional Settings', 'holiday_additional_salary_component')
+        if not salary_component:
+            frappe.throw(_("Please Contact HRD to configure Salary Component for Holiday Additional Salary !!"))
+
+        # create and submit additional salary and compensatory leave request on attendance submit
+        if method == 'on_submit':
+            remark = _("Worked on {0}".format(holidays[0].description))
+            leave_type = frappe.db.get_single_value('HR and Payroll Additional Settings', 'holiday_compensatory_leave_type')
+            if not leave_type:
+                frappe.throw(_("Please Contact HRD to configure Leave Type for Holiday Compensatory Leave Request !!"))
+
+            if not is_site_allowance_exist_for_this_employee(doc.employee, doc.attendance_date):
+                create_additional_salary_from_attendance(doc, salary_component, remark)
+            create_compensatory_leave_request_from_attendance(doc, leave_type, remark)
+
+        # cancel additional salary and compensatory leave request on attendance cancel
+        if method == "on_cancel":
+            cancel_additional_salary_from_attendance(doc, salary_component)
+            cancel_compensatory_leave_request_from_attendance(doc)
+
+def is_site_allowance_exist_for_this_employee(employee, date):
+    '''
+        Check if site allowance exists for the employee in the date
+        return bool
+    '''
+    # Get Employee Schedule for this employee and date
+    employee_schedule = frappe.db.exists('Employee Schedule',
+        {'employee': employee, 'date': date, 'employee_availability': 'Working'})
+    if employee_schedule:
+        # Get Operations Site from the Employee Schedule
+        operations_site = frappe.db.get_value('Employee Schedule', employee_schedule, 'site')
+        if operations_site:
+            # Return include site allowance, then return true
+            return frappe.db.get_value('Operations Site', operations_site, 'include_site_allowance')
```

Comment:
This will return the value (include_site_allowance). It won't return "True"!!

Reply:
@Amna-onefm `include_site_allowance` is a check field in Operations Site, It will return 0 or 1 (False or True)

Reply:
@pjamsheer , Oh Okay, I didn't check that.